src/custom/pages/Swap/SwapMod.tsx

```diff
@@ -289,50 +337,59 @@ export default function Swap({
               : 'Swap w/ Send',
           label: [
             trade?.inputAmount?.currency?.symbol,
-            trade?.outputAmount?.currency?.symbol
-            // getTradeVersion(trade)
-          ].join('/')
-        })
-
-        ReactGA.event({
-          category: 'Routing',
-          action: singleHopOnly ? 'Swap with multihop disabled' : 'Swap with multihop enabled'
+            trade?.outputAmount?.currency?.symbol,
+            // getTradeVersion(trade),
+            singleHopOnly ? 'SH' : 'MH',
+          ].join('/'),
         })
       })
-      .catch(error => {
+      .catch((error) => {
         setSwapState({
           attemptingTxn: false,
           tradeToConfirm,
           showConfirm,
           swapErrorMessage: error.message,
-          txHash: undefined
+          txHash: undefined,
         })
       })
   }, [
-    // priceImpactWithoutFee,
+    priceImpact,
     swapCallback,
     tradeToConfirm,
     showConfirm,
     recipient,
     recipientAddress,
     account,
     trade,
-    singleHopOnly
+    singleHopOnly,
   ])
 
   // errors
   const [showInverted, setShowInverted] = useState<boolean>(false)
 
-  // warnings on slippage
-  // const priceImpactSeverity = warningSeverity(priceImpactWithoutFee)
+  // warnings on the greater of fiat value price impact and execution price impact
+  // const priceImpactSeverity = useMemo(() => {
+  //   const executionPriceImpact = trade?.priceImpact
+  //   return warningSeverity(
+  //     executionPriceImpact && priceImpact
+  //       ? executionPriceImpact.greaterThan(priceImpact)
+  //         ? executionPriceImpact
+  //         : priceImpact
+  //       : executionPriceImpact ?? priceImpact
+  //   )
+  // }, [priceImpact, trade])
+
+  const isArgentWallet = useIsArgentWallet()
 
   // show approve flow when: no error on inputs, not approved or pending, or approved in current session
   // never show if price impact is above threshold in non expert mode
   const showApproveFlow =
+    // TODO: review this
+    !isArgentWallet &&
```

Comment:
I think this has to do with token approval without a tx, or whatever that is.
The one where the token supports allowance natively, and I guess by this point only argent has that implemented?

Reply:
yeah i need to look more into isArgentWallet